packages/crawler/src/page-processors/page-processor-base.ts

```diff
@@ -68,15 +71,35 @@ export abstract class PageProcessorBase implements PageProcessor {
      */
     public gotoFunction: Apify.PuppeteerGoto = async (inputs: Apify.PuppeteerGotoInputs) => {
         try {
-            await inputs.page.setBypassCSP(true);
-
-            return await this.gotoExtended(inputs.page, inputs.request, {
-                waitUntil: 'networkidle0',
-                timeout: this.gotoTimeoutSecs * 1000,
-            });
+            // Configure browser's page settings before navigating to URL
+            await this.pageConfigurator.configurePage(inputs.page);
+
+            let response: Response;
+            try {
+                response = await this.gotoExtended(inputs.page, inputs.request, {
+                    waitUntil: 'networkidle0',
+                    timeout: this.gotoTimeoutSecs * 1000,
+                });
+                // Catch only URL navigation error here
+            } catch (err) {
+                const navigationError = this.pageResponseProcessor.getNavigationError((err as Error).message);
```

Comment:
Let's keep the error trace in the browserError

Reply:
ok